resources/gcp/environment.go

```diff
@@ -54,6 +61,35 @@ func NewEnvironment(ctx *pulumi.Context) (Environment, error) {
 	return env, nil
 }
 
+func logIn(ctx *pulumi.Context) {
+	// Don't log in if the env variables are already set
+	envVariablesSet := false
+	for _, envVar := range pulumiEnvVariables {
+		if os.Getenv(envVar) != "" {
+			fmt.Printf("The env variable %s is set\n", envVar)
```

Comment:
Very good idea!

Reply:
I read the docs again and this env variable can contain either the path to a file that contains the secret or the secret itself directly. To avoid leaking it in this case, I won't print it to stay on the safe side